### PR TITLE
feat: Move My Expenses To Platform: Added Route Guard for redirection And Updated Model

### PR DESCRIPTION
**Pull request by jayfyle**
### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at a39cb71</samp>

This pull request adds a new guard for the `my-expenses-beta` route, which checks the organization settings and redirects the user if the beta feature is disabled. It also adds a test file and two new properties to the `OrgSettings` interface.

### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at a39cb71</samp>

> _To access the beta route of expenses_
> _You need a guard that checks some defenses_
> _It uses `OrgSettingsService`_
> _And `Router` with finesse_
> _To redirect or allow as the case commences_

### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at a39cb71</samp>

*  Add a new guard class `MyExpensesGuardGuard` to check the organization settings and redirect the user if the beta feature is disabled ([link](https://github.com/fylein/fyle-mobile-app/pull/2607/files?diff=unified&w=0#diff-37bc87a5f54c23939dfa87b5be91b770f5995a9dad64d0d4d67780283e6c8e50R1-R23), [link](https://github.com/fylein/fyle-mobile-app/pull/2607/files?diff=unified&w=0#diff-c93d84613627aaadd08e7a7f1d5b20cde2542aacddaf14d50ed5a0676b733036R3), [link](https://github.com/fylein/fyle-mobile-app/pull/2607/files?diff=unified&w=0#diff-c93d84613627aaadd08e7a7f1d5b20cde2542aacddaf14d50ed5a0676b733036R13))
*  Add two new properties to the `OrgSettings` interface to store the beta feature flags ([link](https://github.com/fylein/fyle-mobile-app/pull/2607/files?diff=unified&w=0#diff-7ab4c682c94025cf52d451f9d87c6125129219f7ed1fa32a0ffe24df56d4b38fR554-R555))
*  Add a test file for the `MyExpensesGuardGuard` class and a simple test case to verify its creation ([link](https://github.com/fylein/fyle-mobile-app/pull/2607/files?diff=unified&w=0#diff-e2a717390f614f5357a07f9c56e226af10c8ae8bfb2cef1bfbf27e7ea1ee8e1dR1-R16))

## Clickup
https://app.clickup.com/t/86ctv3kk4

## Code Coverage
Please add code coverage here

## UI Preview
Please add screenshots for UI changes

### CODE REVIEW

File: src/app/core/guards/my-expenses-guard.guard.spec.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { TestBed } from '@angular/core/testing';
+
+import { MyExpensesGuardGuard } from './my-expenses-guard.guard';
+
+describe('MyExpensesGuardGuard', () => {
```
Comment (OmkarJ13) [inline src/app/core/guards/my-expenses-guard.guard.spec.ts:5]: Tests will be in a separate PR?

File: src/app/core/guards/my-expenses-guard.guard.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+})
+export class MyExpensesGuardGuard implements CanActivate {
+  constructor(private orgSettingsSerivce: OrgSettingsService, private router: Router) {}
+
+  canActivate(
+    route: ActivatedRouteSnapshot,
+    state: RouterStateSnapshot
+  ): Observable<boolean | UrlTree> | Promise<boolean | UrlTree> | boolean | UrlTree {
+    this.orgSettingsSerivce.get().subscribe((orgSettings) => {
+      if (!orgSettings?.mobile_app_my_expenses_beta_enabled) {
```
Comment (OmkarJ13) [inline src/app/core/guards/my-expenses-guard.guard.ts:17]: Can remove the optional check here, orgSettings should be present if the api call succeds

File: src/app/core/guards/my-expenses-guard.guard.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+        this.router.navigate(['/', 'enterprise', 'my_expenses']);
+      }
+    });
+    return true;
```
Comment (OmkarJ13) [inline src/app/core/guards/my-expenses-guard.guard.ts:21]: This will always return true no? Should we do this inside the observable?

**Review by OmkarJ13 — COMMENTED** [review]
Some questions

File: src/app/core/guards/my-expenses-guard.guard.spec.ts (hunk 1 above)
Comment (jayfyle) [reply src/app/core/guards/my-expenses-guard.guard.spec.ts]: Added tests

**Review by Dimple16 — APPROVED** [review]

**Review by OmkarJ13 — APPROVED** [review]
Remove fdescribe before merging

**Pull request merged**